Add raw data filter regex and task

paridhi-parajuli opened the pull request:
We need an additional task to filter the discovered files for cases when we need to select specific subfolders or files within the provided prefix. 

# Changes:

- Added raw data regex to the config 
- Added raw data filtering based on the regex
- Make the xcom to pass the location to discovered files in s3 rather than the list itself
- Increase the run per DAG to 20 for `process_files`

Example scenario:
```
{
    "collection_name": "",
    "data_acquisition_method": "s3",
    "data_prefix": "",
    "dest_data_bucket": "ghgc-data-store-develop",
    "ext": ".nc",
    "nodata": -9999,
    "plugins_uri": "",
    "raw_data_bucket": "noaa-goes16",
    "raw_data_prefix": "ABI-L1b-RadF/2024",
    "raw_data_filter_regex": "(178|179)/00/.*MC02.*\.nc$"
}


Say we want the file hierarchy:

2024
├── 178
│   ├── 00
│   │   ├── file1_MC02.nc
│   │   ├── file2_MC08.nc
│   │   └── file3_MC02.nc
│   ├── 01
│   │   ├── file4_MC08.nc
│   │   └── file5_MC02.nc
├── 179
│   ├── 00
│   │   ├── file8_MC02.nc
│   │   └── file9_MC08.nc
│   ├── 01
│   │   ├── file10_MC08.nc
│   │   └── file11_MC02.nc
└── 180
    ├── 00
    │   ├── file14_MC02.nc
    │   └── file15_MC08.nc
    ├── 01
    │   ├── file16_MC08.nc
    │   └── file17_MC02.nc
    
```
    
We want the subfolders `/178` `/179` and, within that just `/00` and only the files with `MC02`

paridhi-parajuli committed:
add raw data filter regex and task

paridhi-parajuli committed:
add config param as str

amarouane-ABDELHAK commented:
Can we increase the run per DAG to like 20? 
This line https://github.com/NASA-IMPACT/veda-data-airflow/blob/automated-transformation/add-raw-data-filter/dags/automated_transformation/automation_dag.py#L117

paridhi-parajuli committed:
increase runs per dag

paridhi-parajuli committed:
1. Combine filter task to discover files task
2. Make xcom to pass the s3 location of discovered files rather than the list itself
3. Do chunking for writing the files to s3 if its more than 900

amarouane-ABDELHAK commented on dags/automated_transformation/automation_dag.py:
```diff
@@ -5,6 +5,9 @@
 from airflow.operators.dummy_operator import DummyOperator
 from slack_notifications import slack_fail_alert
+from airflow.models.variable import Variable
+from dags.veda_data_pipeline.utils.xcom_to_s3 import write_xcom_to_s3,read_xcom_from_s3
```
This import shouldn't work! try 
```code
from veda_data_pipeline.utils.xcom_to_s3 import write_xcom_to_s3,read_xcom_from_s3
```

paridhi-parajuli committed:
resolve comment

amarouane-ABDELHAK commented on dags/veda_data_pipeline/utils/xcom_to_s3.py:
```diff
@@ -0,0 +1,16 @@
+def read_xcom_from_s3(s3_uri):
+    with smart_open.open(s3_uri, "r") as _file:
+        s3_event_read = _file.read()
+    return json.loads(s3_event_read)
```
Add new line here

amarouane-ABDELHAK commented on dags/automated_transformation/automation_dag.py:
```diff
@@ -91,18 +96,42 @@ def discover_files(ti):
+        return chunks_xcom
+
+
+    @task(max_active_tis_per_dag=20)
```
We probably need to make this value configurable to limit the parallel runs, so we can prevent a full redeployment to just run less or more tasks.

paridhi-parajuli committed:
add new line

paridhi-parajuli committed:
make the max instances per dag configurable

paridhi-parajuli committed:
add changes to temp file handling
